scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py

```diff
@@ -142,45 +200,95 @@ def plot_dump(
-    if (args.tc is not None) and (args.vc is not None):
+    logger.setLevel(args.log_level)
+
+    if args.tc and args.vc:
         raise ValueError(
             "Only one of --tensor-component and --vector-component should be set."
         )
+
+    if args.workers > 1:
+        logger.warning(
+            "Matplotlib is not multi-thread friendly. Read this for more details https://matplotlib.org/stable/users/faq/howto_faq.html#work-with-threads"
+        )
+        logger.warning(
+            "Try decreasing threads if you encounter any undefined behaviour"
+        )
+    # Create output director if does't exists
+    args.output_directory.mkdir(0o755, True, True)
+    logger.info(f"Total files to process: {len(args.files)}")
+
+    components = [0, 0]
     if args.tc is not None:
         components = args.tc
     if args.vc is not None:
         components = [0, args.vc]
-    dump_id = 0
-    debug_plot = False
-    for f in files:
-        data = read(f)
-        print(data)
-        q = data.Get(field, False, not debug_plot)
-        name = str(dump_id).rjust(4, "0") + ".png"
-        if debug_plot:
-            plot_dump(
-                data.xg,
-                data.yg,
-                q,
-                name,
-                True,
-                data.gid,
-                data.xig,
-                data.yig,
-                data.xeg,
-                data.yeg,
-                components,
+
+    _x = ProcessPoolExecutor if args.worker_type == "process" else ThreadPoolExecutor
+    with _x(max_workers=args.workers) as pool:
+        for dump_id, file_name in enumerate(args.files):
+            data = phdf(file_name)
+            if args.field not in data.Variables:
+                logger.error(
+                    f'No such field "{args.field}" in {file_name}. This will lead to stop further processing'
+                )
+                logger.info(f"Available fields: {data.Variables}")
+                ERROR_FLAG = True
+                break
+
+            q = data.Get(args.field, False, not args.debug)
+            name = "{}_{:04d}.png".format(args.prefix, dump_id).strip()
+            output_file = args.output_directory / name
+
+            # NOTE: After doing 5 test on different precision, keeping 2 looks more promising
+            current_time = format(round(data.Time, 2), ".2f")
+            if args.debug:
+                pool.submit(
+                    plot_dump,
+                    data.xg,
+                    data.yg,
+                    q,
+                    current_time,
+                    output_file,
+                    True,
+                    data.gid,
+                    data.xig,
+                    data.yig,
+                    data.xeg,
+                    data.yeg,
+                    components,
+                )
+            else:
+                pool.submit(
+                    plot_dump, data.xng, data.yng, q, current_time, output_file, True
+                )
```

Comment:
Maybe I am missing something, but it looks like `components` is not getting passed here. I think that was maybe true of the old code as well, but I think it was a bug.

Reply:
It is not even passed in the develop branch. This is the latest commit in the develop branch.

https://github.com/parthenon-hpc-lab/parthenon/blob/a61999403a09a2946bd6ae40b5140169833b844a/scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py#L185

Reply:
I think what @lroberts36 is saying is that this is a bug in `develop`

Reply:
I see, should I add it then?

Reply:
assume it doesn't break functionality, yes please.

Reply:
yes, I think so

Reply:
@Yurlungur

Done, sir